### PR TITLE
add resourcelock, leaderelection for cluster-policy-controller

### PR DESCRIPTION
**Pull request by sallyom**
I've added a leader lock in the NamespaceSCCAllocationController for upgrades 4.2->4.3 
to avoid conflicts with cluster-policy-controller container running in kube-controller-manager static pod

@deads2k 
@stlaz

### CODE REVIEW

File: pkg/cmd/openshift-controller-manager/controller_manager.go [hunk 1]
```diff
@@ -74,40 +74,54 @@ func RunOpenShiftControllerManager(config *openshiftcontrolplanev1.OpenShiftCont
...
-		leaderelection.LeaderElectionConfig{
-			Lock:          rl,
-			LeaseDuration: config.LeaderElection.LeaseDuration.Duration,
-			RenewDeadline: config.LeaderElection.RenewDeadline.Duration,
-			RetryPeriod:   config.LeaderElection.RetryPeriod.Duration,
-			Callbacks: leaderelection.LeaderCallbacks{
-				OnStartedLeading: originControllerManager,
-				OnStoppedLeading: func() {
-					klog.Fatalf("leaderelection lost")
+	components := []string{"openshift-controller-manager", "cluster-policy-controller"}
```
Comment (deads2k) [inline pkg/cmd/openshift-controller-manager/controller_manager.go]: no.  Only do a second leader election on the SCC namespace controller in particular, not all of it.

Comment (sallyom) [reply pkg/cmd/openshift-controller-manager/controller_manager.go]: ok

Comment (sallyom) [reply pkg/cmd/openshift-controller-manager/controller_manager.go]: I've updated, i'm new to this code, hopefully it's close, with cluster-admin rbac (will figure that out next)  i get the leader lock cm in cluster-kube-controller-manager-ns.

**Comment by sallyom** [comment]
clusters failing with `error retrieving resource lock openshift-kube-controller-manager/cluster-policy-controller: configmaps "cluster-policy-controller" is forbidden: User "system:serviceaccount:openshift-infra:namespace-security-allocation-controller" cannot get resource "configmaps" in API group "" in the namespace "openshift-kube-controller-manager"` from openshift-controller-manager pod logs. 

The rolebinding needed is [here](openshift/cluster-openshift-controller-manager-operator#120), and the role is [here](https://github.com/openshift/cluster-kube-controller-manager-operator/pull/287).

**Comment by deads2k** [comment]
open against master first please

**Comment by sallyom** [comment]
closing in favor of https://github.com/openshift/openshift-controller-manager/pull/33

**Pull request closed without merging**